src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -36,19 +38,29 @@ class HttpKernel implements HttpKernelInterface, TerminableInterface
     protected $dispatcher;
     protected $resolver;
     protected $requestStack;
+    private $argumentResolver;
 
     /**
      * Constructor.
      *
-     * @param EventDispatcherInterface    $dispatcher   An EventDispatcherInterface instance
-     * @param ControllerResolverInterface $resolver     A ControllerResolverInterface instance
-     * @param RequestStack                $requestStack A stack for master/sub requests
+     * @param EventDispatcherInterface    $dispatcher       An EventDispatcherInterface instance
+     * @param ControllerResolverInterface $resolver         A ControllerResolverInterface instance
+     * @param RequestStack                $requestStack     A stack for master/sub requests
+     * @param ArgumentResolverInterface   $argumentResolver An ArgumentResolverInterface instance
      */
-    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null)
+    public function __construct(EventDispatcherInterface $dispatcher, ControllerResolverInterface $resolver, RequestStack $requestStack = null, ArgumentResolverInterface $argumentResolver = null)
```

Comment:
The constructor is the only method that I know of which is allowed to have a different signature in child classes. If the constructor is overridden, it will simply fall back to the same logic that is triggered in the old situation. It will pass the `ControllerResolver` into both properties and this will make sure it triggers a deprecation.